roles/haproxy/templates/haproxy_frontend.cfg.j2

```diff
@@ -106,10 +119,17 @@ frontend localhost_restricted
     option httplog
     capture request header User-agent len 256
     capture request header X-TLS-Client len 256
-    # The following lines are to log the request rate per 10s
-    stick-table type ip size 1m expire 10s store http_req_rate(10s)
-    http-request track-sc0 src
+    # Allowlist ACL
+    acl allowlist src -f /etc/haproxy/acls/allowedips.acl
+    # Measure and log the request rate per 10s
+    http-request track-sc0 src table st_httpreqs_per_ip
     http-request capture sc_http_req_rate(0) len 4
+    # Create an ACL when the request rate exceeds {{ haproxy_max_request_rate }} per 10s and deny them
+    acl exceeds_max_request_rate_per_ip sc_http_req_rate(0) gt {{ haproxy_max_request_rate }}
+    http-request deny deny_status 429 if exceeds_max_request_rate_per_ip !allowlist
```

Comment:
How useful is it to apply these things to vhosts only accessible from trusted networks?

Reply:
If they're trusted, and well behaved, there is no issue. If they do generate requests in these numbers, I'd say they should be blocked as well